node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -4152,6 +4463,36 @@ mod tests {
             .for_each(|n| db.node_by_key_mut(n.public_key()).unwrap().resign());
     }
 
+    fn make_ipchange(
+        n: u16,
+        old_addr_opt: Option<NodeAddr>,
+        new_addr_opt: Option<NodeAddr>,
+        mode: Mode,
+    ) -> (
+        Gossip_0v1,
+        NodeRecord,
+        NeighborhoodDatabase
+    ) {
+        let mut new_ipchange_node = make_node_record(n, true);
+        let mut old_ipchange_node = new_ipchange_node.clone();
+        new_ipchange_node.metadata.node_addr_opt = new_addr_opt.clone();
+        old_ipchange_node.metadata.node_addr_opt = old_addr_opt.clone();
```

Comment:
This `old_addr_opt` in particular might not need be cloned you can move the ownership directly.

Reply:
Good catch.